default.nix

```diff
@@ -0,0 +1,49 @@
+#
+# This file is just a script to setup the environment using Nix and is not typically used
+# in practice; a more well known alternative for environment setup and packaging can be found
+# in the Dockerfile. For more information see https://nixos.org/nix/ and
+# https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.md
+#
```

Comment:
Per our discussion, let's keep this out of master/develop for now -- orphan branch is fine.